plenoptic/metric/perceptual_distance.py

```diff
@@ -430,11 +449,22 @@ def nlpd(IM_1, IM_2):
        assessment using a normalized Laplacian pyramid. Electronic Imaging, 2016(16), pp.1-6.
     """
 
-    if IM_1.shape[1] > 1 or IM_2.shape[1] > 1:
-        warnings.warn("NLPD was developed on grayscale images, so the channel dimension "
-                      "is treated as another batch dimension.")
-    y1 = normalized_laplacian_pyramid(IM_1)
-    y2 = normalized_laplacian_pyramid(IM_2)
+    if not img1.ndim == img2.ndim == 4:
+        raise Exception("Input images should have four dimensions: (batch, channel, height, width)")
+    if img1.shape[-2:] != img2.shape[-2:]:
+        raise Exception("img1 and img2 must have the same height and width!")
+    for i in range(2):
+        if img1.shape[i] != img2.shape[i] and img1.shape[i] != 1 and img2.shape[i] != 1:
+            raise Exception("Either img1 and img2 should have the same number of "
+                            "elements in each dimension, or one of "
+                            "them should be 1! But got shapes "
+                            f"{img1.shape}, {img2.shape} instead")
+    if img1.shape[1] > 1 or img2.shape[1] > 1:
+        warnings.warn("NLPD was designed for grayscale images and here it will be computed separately for each "
+                      "channel (so channels are treated in the same way as batches).")
+    
```

Comment:
given that this is a check we do regularly, might be worth writing a helper function for it. doesn't necessarily need to be in this PR though

Reply:
if you decide not to put it in this PR, can you create an issue for this? and paste your code in there.

Reply:
created issue in #185